InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java

```diff
@@ -127,4 +133,29 @@ public void rotateToDeg(double targetPoint) {
         targetChange = 0;
     }
 
-}
\ No newline at end of file
+    public void rotateByDeg(double degrees) {
+        rotateToDeg(getTargetTurretDegGivenOffset(degrees));
+    }
+    
+    public double getTurretDeg() {
+        double encoderDeg = MathUtils.unitConverter(azimuthMotor.getSelectedSensorPosition(), config.shooter.azimuth.ticksPerRevolution, 360.0);
+        double turretDeg = encoderDeg * config.shooter.gearRatio;
+        return turretDeg;
+    }
+
+    /*
+        Returns target degrees of turret given an offset
+        Assuming both sides agree that "left" is a negative offset.
+    */
+    public double getTargetTurretDegGivenOffset(double offset) {
+        return getTurretDeg() + offset;
+    }
+
+    public void rotateTurretGivenLLOffset(double offset) {
+        // For LL, "left" is a negative offset. For the turret, "left" is a positive offset. For this reason, we have to subtract the offset instead of adding it.
+        // Calculate the average of the last values passed in (up to 25) in order to provide a more accurate number
+        double avgDegrees = filter.calculate(getTurretDeg() - offset);

```

Comment:
yay looks good! however, could I convince you to also add another boolean to the shootersubsystem representing whether to use a filter or not? because we don't actually know as of right now, first we'll test without and then with if needed. so something like:
```Java
degrees = getTurretDeg() - offset
if (ShooterConstants.USE_FILTER) {
    degrees = filter.calculate(degrees);
}
...
```
or similar? also maybe add a short explanation of why we may want to use the filter in comments for future generations of the team